src/Compatibility/Core/src/Windows/FrameworkElementExtensions.cs

```diff
@@ -77,22 +74,7 @@ static DependencyProperty GetForegroundProperty(FrameworkElement element)
 			if (element is TextBlock)
 				return TextBlock.ForegroundProperty;
 
-			Type type = element.GetType();
-
-			DependencyProperty foregroundProperty;
-			if (!ForegroundProperties.Value.TryGetValue(type, out foregroundProperty))
-			{
-				FieldInfo field = ReflectionExtensions.GetFields(type).FirstOrDefault(f => f.Name == "ForegroundProperty");
-				if (field == null)
-					throw new ArgumentException("type is not a Foregroundable type");
-
-				var property = (DependencyProperty)field.GetValue(null);
-				ForegroundProperties.Value.TryAdd(type, property);
-
-				return property;
-			}
-
-			return foregroundProperty;
+			throw new InvalidOperationException($"Cannot get ForegroundProperty of {element.GetType()}");
```

Comment:
@jonathanpeppers @PureWeen this code will be removed in https://github.com/dotnet/maui/pull/21238. I suggest merging this PR now and cleaning up/removing this Windows-specific code later.